velox/exec/HashPartitionFunction.h

```diff
@@ -58,4 +58,31 @@ class HashPartitionFunction : public core::PartitionFunction {
   SelectivityVector rows_;
   raw_vector<uint64_t> hashes_;
 };
+
+class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
+ public:
+  HashPartitionFunctionSpec(
```

Comment:
Let's add some documentation to explain the relationship between constValues and keys. I assume constValues.size must be equal to keys.size and some values can be null, while values that correspond to kConstantChannel must be not null. Let's also add sanity checks to enforce this.